ts/utils/navigation.ts

```diff
@@ -59,12 +75,14 @@ export function getRouteName(route: NavigationRoute): Option<string> {
 }
 
 /**
- * This function returns the name of the precedent navigation route to understand
- * from where the current route has been navigated
+ * @deprecated
  */
-export function whereAmIFrom(nav: NavigationHistoryState): Option<string> {
-  const navLength = nav.length;
-  return index(navLength - 1, [...nav]).fold(none, ln =>
-    getRouteName(ln.routes[ln.index])
-  );
-}
+export const isOnboardingCompleted = () => {
+  const route = NavigationService.getCurrentState();
+  if (route === null) {
+    return false;
+  }
+  return route !== undefined && route.routes !== undefined
+    ? route.routes.length > 0 && route.routes[0].routeName === ROUTES.MAIN
+    : false;
```

Comment:
```suggestion
  return route.routes?.[0].routeName === ROUTES.MAIN;
```
Could be a good refactoring?

Reply:
Changed in 4202ad6a46de8d68e6f5f5473c66a01c8b3d73a4:
`route.routes?.length > 0 && route.routes[0].routeName === ROUTES.MAIN;`

we need to check the size because if `routes.length === 0` we have an exception.

Reply:
in JS if you try to access a position in an array that doesn't exists, it returns `undefined` :D
https://replit.com/@MatteoPescasub/GargantuanWigglyLinuxkernel#index.js
btw consider it as resolved!

Reply:
Yes but in this case is not the `undefined` when try to access to the array the problem, but trying to use `.routeName` to an `undefined` object. 
https://www.typescriptlang.org/play?#code/C4TwDgpgBAIghsOUC8UDeAoK2oDMCWEANgCYBcUAzsAE74B2A5gNwYC+GGREwUJCcAII0acEBQBKEOCQD29IiGGiQAHniIAfAB8oAV3okIBehBKsM+XAAoAsggAWAOlGHZAW2sBKKJqgAGLwxMHD4BZTEUKABtAF1WDgwAY3lKWW4nIllGa35ECJAAfido-1inAmISZBqAIiT8OFlar2YgA

In a second thought can be changed in:
`route.routes?.[0]?.routeName === ROUTES.MAIN;`
I hadn't thought of it before.

Reply:
`routeName` could be raise an error
this is right, I didn't spot that case!